Searching Algorithms/Binary Search Algorithm/C++/binary-search.cpp

```diff
@@ -4,66 +4,73 @@ Theory of Programming
 
 Binary Search Algorithm
 http://theoryofprogramming.com/2016/11/04/binary-search-algorithm/
-GitHub - https://github.com/salman-bhai/theoryofprogramming
-Code Contributor - <salman-bhai>
+GitHub - https://github.com/VamsiSangam/theoryofprogramming
+Code Contributors - <salman-bhai>
+                  - Praveen Mylavarapu (github.com/praveenmylavarapu, saipraveen.ml)
 
 ===== ===== ===== */
-// Includes all header files required for C++ functions
-#include <bits/stdc++.h>
-using namespace std;
 
-// Variable declaration
-vector<int> v;
+#include <iostream>
+#include <vector>
+using namespace std;
 
 // Main Binary Search logic
-int binary(int key) {
+int binarySearch(vector<int> v, int val) {
 	int first = 0;
 	int last = v.size() - 1;
 
 	while(first <= last) {
 		int mid = ((first + last) / 2);
```

Comment:
Is it really necessary to make it more complicated? Because one doesn't enter such large input for overflow to occur.